reference/conanfile/methods/finalize.rst

```diff
@@ -0,0 +1,74 @@
+.. _reference_conanfile_methods_finalize:
+
+finalize()
+==========
+
+.. include:: ../../../common/experimental_warning.inc
+
+Package immutability is an important concept in Conan. It ensures that the package is not modified after it has been built and packaged,
+so that the package id is consistent and the package can be reused in different machines.
+
+This method is intended for customization of the package in the running machine, allowing modifications to the package
```

Comment:
I'd say this would need a bit more explanation/setup for the problem/approach to fix it, wdyt?